src/snowflake/snowpark/udf.py

```diff
@@ -66,21 +70,32 @@ class UserDefinedFunction:
         - :func:`~snowflake.snowpark.functions.udf`
     """
 
+    # def __init__(
+    #     self,
+    #     func: Union[Callable, Tuple[str, str]],
+    #     return_type: DataType,
+    #     input_types: List[DataType],
+    #     name: str,
+    #     is_return_nullable: bool = False,
+    # ) -> None:
+    #     #: The Python function or a tuple containing the Python file path and the function name.
+    #     self.func: Union[Callable, Tuple[str, str]] = func
+    #     #: The UDF name.
+    #     self.name: str = name
+
+    #     self._return_type = return_type
+    #     self._input_types = input_types
+    #     self._is_return_nullable = is_return_nullable
+
     def __init__(
         self,
-        func: Union[Callable, Tuple[str, str]],
-        return_type: DataType,
-        input_types: List[DataType],
-        name: str,
+        callableProperties: CallableProperties,
```

Comment:
Technically UserDefinedFunction is a public type, which makes this a backwards-incompatible change. Check with Snowpark whether they're ok changing this ctor. Otherwise, you'd want to add anything you need after the last parameter, and make sure there is a default value assigned.

Reply:
That, or to keep this public class as is (and other classes too like SPROC, UDxF), the UDF creation in udf.py can be changed to accept the specific instance variables of `callableProperties` instead of passing in the entire object itself.